src/main/java/ecdar/backend/SimulationHandler.java

```diff
@@ -0,0 +1,478 @@
+        // j is needed to map the correct name with the value
+//        for (int i = noOfVars, j = 0; i < noOfClocks + noOfVars ; i++, j++) {
+//            simulationClocks.put(engine.getSystem().getClockName(j),
+//                    currentConcreteState.get().getVariables()[i].getValue(BigDecimal.ZERO));
+//        }
+    }
+
+    /**
+     * Getter for the current concrete state
+     *
+     * @return the current {@link SimulationState}
+     */
+    public SimulationState getCurrentState() {
+        return currentConcreteState.get();
+    }
+
+    /**
+     * The way to get the time in the current state of a simulation
+     *
+     * @return the time in the current state
+     */
+    public BigDecimal getCurrentTime() {
+        return currentTime.get();
+    }
+
+    public ObjectProperty<BigDecimal> currentTimeProperty() {
+        return currentTime;
+    }
+
+    /**
+     * The way to get the delay of the latest step in the simulation
+     *
+     * @return the delay of the latest step in the in the simulation
+     */
+    public BigDecimal getDelay() {
+        return delay;
+    }
+
+    /**
+     * The number of total steps taken in the current simulation
+     *
+     * @return the number of steps
+     */
+    public int getNumberOfSteps() {
+        return numberOfSteps;
+    }
+
+    /**
+     * All the transitions taken in this simulation
+     *
+     * @return an {@link ObservableList} of all the transitions taken in this simulation so far
+     */
+    public ObservableList<SimulationState> getTraceLog() {
+        return traceLog;
+    }
+
+    /**
+     * All the available transitions in this state
+     *
+     * @return an {@link ObservableList} of all the currently available transitions in this state
+     */
+    public ObservableList<ecdar.simulation.Transition> getAvailableTransitions() {
+        return availableTransitions;
+    }
+
+    /**
+     * All the variables connected to the current simulation.
+     * This does not return any clocks, if you need please use {@link SimulationHandler#getSimulationClocks()} instead
+     *
+     * @return a {@link Map} where the name (String) is the key, and a {@link BigDecimal} is the value
+     */
+    public ObservableMap<String, BigDecimal> getSimulationVariables() {
+        return simulationVariables;
+    }
+
+    /**
+     * All the clocks connected to the current simulation.
+     *
+     * @return a {@link Map} where the name (String) is the key, and a {@link BigDecimal} is the clock value
+     * @see SimulationHandler#getSimulationVariables()
+     */
+    public ObservableMap<String, BigDecimal> getSimulationClocks() {
+        return simulationClocks;
+    }
+
+    /**
+     * The initial state of the current simulation
+     *
+     * @return the initial {@link SimulationState} of this simulation
+     */
+    public SimulationState getInitialConcreteState() {
+        // ToDo: Implement
+        return initialConcreteState.get();

```

Comment:
Does this really work? To me it seems like this should construct a `SimulationState`, not try to get the state from the `SimpleObjectProperty` which would be empty because it is never set to an actual `SimulationState`.

Reply:
Done - fjernet metode da den ikke bliver brugt

Reply:
Jeg kan se at den her metode er det samme, er der nogen opdatering på det?

Reply:
hmm troede den var slettet. det fixer jeg lige 👍